cmd/session-clear.go

```diff
@@ -0,0 +1,167 @@
+/*
+ * Minio Client (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"encoding/json"
+
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/console"
+	"github.com/minio/minio/pkg/probe"
+)
+
+var sessionClearFlags = []cli.Flag{
+	cli.BoolFlag{
+		Name:  "force",
+		Usage: "Force a dangerous clear operation.",
+	},
+}
+
+var sessionClear = cli.Command{
+	Name:            "clear",
+	Usage:           "Clear a previously saved session.",
+	Action:          mainSessionClear,
+	Before:          setGlobalsFromContext,
+	Flags:           append(sessionClearFlags, globalFlags...),
+	HideHelpCommand: true,
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+  
+USAGE:
+  {{.HelpName}} SESSION-ID|all
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. Clear session.
+	 $ {{.HelpName}} clear ygVIpSJs
+  2. Clear all sessions.
+	 $ {{.HelpName}} clear all
+  3. Forcefully clear a useless session.
+	 $ {{.HelpName}} --force
+`,
+}
+
+// clearSessionMessage container for clearing session messages.
+type clearSessionMessage struct {
+	Status    string `json:"success"`
+	SessionID string `json:"sessionId"`
+}
+
+// String colorized clear session message.
+func (c clearSessionMessage) String() string {
+	msg := "Session `" + c.SessionID + "`"
+	var colorizedMsg string
+	switch c.Status {
+	case "success":
+		colorizedMsg = console.Colorize("ClearSession", msg+" cleared successfully.")
+	case "forced":
+		colorizedMsg = console.Colorize("ClearSession", msg+" cleared forcefully.")
+	}
+	return colorizedMsg
+}
+
+// JSON jsonified clear session message.
+func (c clearSessionMessage) JSON() string {
+	clearSessionJSONBytes, e := json.Marshal(c)
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+
+	return string(clearSessionJSONBytes)
+}
+
+// forceClear - Remove a saved session.
+// Used if --force flag is applied.
+func forceClear(sid string, session *sessionV8) {
+	if session != nil {
+		if err := session.Delete().Trace(sid); err == nil {
+			// Force unnecesseray removal successful.
+			printMsg(clearSessionMessage{Status: "success", SessionID: sid})
+			return
+		}
+	}
+	// Remove useless session files.
+	removeSessionFile(sid)
+	removeSessionDataFile(sid)
+	printMsg(clearSessionMessage{Status: "forced", SessionID: sid})
+}
+
+// clearSession clear sessions.
+func clearSession(sid string, isForce bool) {
+	if sid == "all" {
```

Comment:
@awwalker, to avoid repetitive code here, you can do the following:

```
var sidsToRemove []string
if sid == "all" {
   sidsToRemove = getSessionIDs()
} else {
   sidsToRemove = append(sidsToRemove, sid)
}

// Now remove all sids in sids To remove

```